src/main/java/com/invoiced/entity/Note.java

```diff
@@ -13,8 +13,12 @@ public Note(Connection conn) {
 
 	public Note(Connection conn, long customerId, long invoiceId) {
 		super(conn, Note.class);
-		this.customerId = customerId;
-		this.invoiceId = invoiceId;
+		if (invoiceId > 0) {
```

Comment:
I am still of the opinion that you should have a setListUrl() method so that the Note class is not aware of the invoice or customer endpoints

Reply:
I thought about this more and agree; I will now be making some changes related to getters and setters to handle sub-entities better.

Reply:
There's a bit more to do to make it modular given contacts and payment plans also have to be handled correctly, but it shouldn't take that long